allennlp/common/file_utils.py

```diff
@@ -226,18 +254,30 @@ def get_from_cache(url: str, cache_dir: str = None) -> str:
     os.makedirs(cache_dir, exist_ok=True)
 
     # Get eTag to add to filename, if it exists.
-    if url.startswith("s3://"):
-        etag = s3_etag(url)
-    else:
-        with session_with_backoff() as session:
-            response = session.head(url, allow_redirects=True)
-        if response.status_code != 200:
-            raise IOError(
-                "HEAD request failed for url {} with status code {}".format(
-                    url, response.status_code
-                )
-            )
-        etag = response.headers.get("ETag")
+    try:
+        if url.startswith("s3://"):
+            etag = _s3_etag(url)
+        else:
+            etag = _http_etag(url)
+    except (ConnectionError, EndpointConnectionError):
+        # We might be offline, in which case we don't want to throw an error
+        # just yet. Instead, we'll try to use the latest cached version of the
+        # target resource, if it exists. We'll only throw an exception if we
+        # haven't cached the resource at all yet.
+        logger.warning(
+            "Connection error occured while trying to fetch etag for %s. "
+            "Will attempt to use latest cached version of resource",
+            url,
+        )
+        latest_cached = _find_latest_cached(url, cache_dir)
+        if latest_cached:
+            return latest_cached
+        logger.error(
```

Comment:
This is a nit, but I prefer to have an `else` clause.  I had trouble reading this in the CR because of this (an how Python handles blocks with spacing)--but admittedly it's just personal preference.

Reply:
Oh yea that's totally fine with me, I'm just used to certain linters raising a fit about "unnecessary else clause". I guess either flake8 doesn't care or we're explicitly ignoring that in our `.flake8`